### PR TITLE
Fix some more commonly found issues

### PR DESCRIPTION
**Pull request by BenCat07**
E.g. snap after closing menu, Projectile aimbot not compensating for ping (Tested to be needed), and Aimbot shooting at walls with projectiles/Not shooting when it should be able to.

### CODE REVIEW

File: src/hacks/Aimbot.cpp [hunk 1]
```diff
@@ -888,8 +888,9 @@ void Aim(CachedEntity *entity)
     if (CE_BAD(entity))
         return;
 
-    // Get angles
-    Vector angles = GetAimAtAngles(g_pLocalPlayer->v_Eye, PredictEntity(entity), LOCAL_E);
+    Vector predicted_origin = PredictEntity(entity);
```
Comment (TotallyNotElite) [inline src/hacks/Aimbot.cpp:891]: Why does this variable exist

File: src/helpers.cpp [hunk 2]
```diff
@@ -1021,14 +1021,21 @@ bool IsEntityVectorVisible(CachedEntity *entity, Vector endpos, unsigned int mas
     if (CE_BAD(entity))
         return false;
     trace::filter_default.SetSelf(RAW_ENT(g_pLocalPlayer->entity));
-    ray.Init(g_pLocalPlayer->v_Eye, endpos);
+    Vector eye = g_pLocalPlayer->v_Eye;
+    // Adjust for weapon offsets if needed
+    if (use_weapon_offset)
+    {
+        auto angle = GetAimAtAngles(g_pLocalPlayer->v_Eye, endpos, LOCAL_E);
```
Comment (TotallyNotElite) [inline src/helpers.cpp:1028]: Why does this variable exist

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]

File: src/hacks/Aimbot.cpp [hunk 3]
```diff
@@ -1171,7 +1171,7 @@ int BestHitbox(CachedEntity *target)
                     headonly = false;
                 // Rocket launcher
             }
-            else if (ci == CL_CLASS(CTFRocketLauncher) || ci == CL_CLASS(CTFRocketLauncher_AirStrike) || ci == CL_CLASS(CTFRocketLauncher_DirectHit) || ci == CL_CLASS(CTFRocketLauncher_Mortar) || ci == CL_CLASS(CTFPipebombLauncher))
+            else if (ci == CL_CLASS(CTFPipebombLauncher) || ci == CL_CLASS(CTFGrenadeLauncher) || ci == CL_CLASS(CTFRocketLauncher) || ci == CL_CLASS(CTFRocketLauncher_AirStrike) || ci == CL_CLASS(CTFRocketLauncher_DirectHit) || ci == CL_CLASS(CTFRocketLauncher_Mortar) || ci == CL_CLASS(CTFPipebombLauncher))
             {
                 preferred = hitbox_t::hip_L;
```
Comment (TotallyNotElite) [inline src/hacks/Aimbot.cpp:1177]: Why exactly should the rocket launcher NOT aim at someone's feet?

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]

**Comment by TotallyNotElite** [comment]
Writing a message here because the GitHub bot refuses to notify anyone about reviews.

File: src/hacks/Aimbot.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hacks/Aimbot.cpp:1177]: Midair we aim at hip due to it being more centered and easier to hit that way

File: src/hacks/Aimbot.cpp (hunk 1 above)
Comment (BenCat07) [reply src/hacks/Aimbot.cpp:891]: Because i forgot to use it

File: src/helpers.cpp (hunk 2 above)
Comment (BenCat07) [reply src/helpers.cpp:1028]: same as above

File: src/hacks/Aimbot.cpp (hunk 3 above)
Comment (TotallyNotElite) [reply src/hacks/Aimbot.cpp:1177]: Uh? Where is the midair check please?

Comment (BenCat07) [reply src/hacks/Aimbot.cpp:1177]: immediately below.

Comment (BenCat07) [reply src/hacks/Aimbot.cpp:1177]:
```            // Airborn projectile
            if (GetWeaponMode() == weaponmode::weapon_projectile)
            {
                if (g_pLocalPlayer->weapon()->m_iClassID() != CL_CLASS(CTFCompoundBow))
                {
                    bool ground = CE_INT(target, netvar.iFlags) & (1 << 0);
                    if (!ground)
                        preferred = hitbox_t::spine_3;
                }
            }```

Comment (BenCat07) [reply src/hacks/Aimbot.cpp:1177]: Wait, why does it aim at hip_l if they are grounded

**Review by TotallyNotElite — APPROVED** [review]

File: src/helpers.cpp [hunk 4]
```diff
@@ -1724,6 +1733,90 @@ void PrintChat(const char *fmt, ...)
...
+        vecOffset = Vector(16.0f, 8.0f, -6.0f);
+        break;
+
+    case CL_CLASS(CTFSyringeGun):
+        vecOffset = Vector(16.0f, 6.0f, -8.0f);
+        break;
+
+    // Huntsman
+    case CL_CLASS(CTFCompoundBow):
+        vecOffset = Vector(23.5f, -8.0f, -3.0f);
```
Comment (acidtripparty) [inline src/helpers.cpp:1780]: How did you obtain the offsets over here? @BenCat07

**Pull request merged**